Typescript errors for go blocks

I am running a minimal typescript setup using npm. I cannot get the types for go blocks to work. Here is the simple code that I'm running and the compile error it generates:

```typescript
import { resolve, go } from 'fluture'

const func = () => go (function* () {
    return yield resolve (1)
})
```

Error:
```
[eval].ts:2:24 - error TS2345: Argument of type '() => Generator<FutureInstance<unknown, number>, number, FutureInstance<L, any>>' is not assignable to parameter of type 'Generator<FutureInstance<unknown, any>, FutureInstance<unknown, number>>'.
  Type 'Generator<FutureInstance<unknown, number>, number, FutureInstance<L, any>>' is not assignable to type 'Iterator<FutureInstance<unknown, any>, FutureInstance<unknown, number>>'.
    Types of property 'next' are incompatible.
      Type '(...args: [] | [FutureInstance<L, any>]) => IteratorResult<FutureInstance<unknown, number>, number>' is not assignable to type '(value?: FutureInstance<unknown, any>) => Next<FutureInstance<unknown, any>> | Done<FutureInstance<unknown, number>>'.
        Type 'IteratorResult<FutureInstance<unknown, number>, number>' is not assignable to type 'Next<FutureInstance<unknown, any>> | Done<FutureInstance<unknown, number>>'.
          Type 'IteratorYieldResult<FutureInstance<unknown, number>>' is not assignable to type 'Next<FutureInstance<unknown, any>> | Done<FutureInstance<unknown, number>>'.
            Type 'IteratorYieldResult<FutureInstance<unknown, number>>' is not assignable to type 'Done<FutureInstance<unknown, number>>'.
              Property 'done' is optional in type 'IteratorYieldResult<FutureInstance<unknown, number>>' but required in type 'Done<FutureInstance<unknown, number>>'.
```

package.json:
```javascript
{
  "name": "ipfs-playground",
  "version": "1.0.0",
  "main": "index.js",
  "license": "MIT",
  "scripts": {
    "start": "npm run build && node dist/app.js",
    "debug": "npm run build && node --nolazy --inspect-brk=0.0.0.0:9229  dist/app.js",
    "build": "tsc"
  },
  "dependencies": {
    "@types/node": "^12.7.3",
    "fluture": "^12.0.0-beta.3",
    "ts-node": "^8.3.0",
    "typescript": "^3.6.2"
  }
}
```
tsconfig.json:
```javascript
{
    "compilerOptions": {
        "module": "commonjs",
        "target": "ES2015",
        "declaration": true,
        "moduleResolution": "node",
        "sourceMap": true,
        "outDir": "dist",
        "strict": true,
        "baseUrl": "."
    }
}
```

--- a/index.d.ts
+++ b/index.d.ts
@@ -1,3 +1,6 @@
+/// <reference lib="es2015.generator" />
+/// <reference lib="es2015.iterable" />
+
 declare module 'fluture' {
 
   export interface RecoverFunction {
@@ -18,24 +21,6 @@ declare module 'fluture' {
 
   export interface Nodeback<E, R> {
     (err: E | null, value?: R): void
-  }
-
-  export interface Next<T> {
-    done: boolean
-    value: T
-  }
-
-  export interface Done<T> {
-    done: boolean
-    value: T
-  }
-
-  export interface Iterator<N, D> {
-    next(value?: N): Next<N> | Done<D>
-  }
-
-  export interface Generator<Y, R> {
-    (): Iterator<Y, R>
   }
 
   export interface ConcurrentFutureInstance<L, R> {
@@ -131,7 +116,7 @@ declare module 'fluture' {
   export function forkCatch(recover: RecoverFunction): <L>(reject: RejectFunction<L>) => <R>(resolve: ResolveFunction<R>) => (source: FutureInstance<L, R>) => Cancel
 
   /** Build a coroutine using Futures. See https://github.com/fluture-js/Fluture#go */
-  export function go<L, R>(generator: Generator<FutureInstance<L, any>, R>): FutureInstance<L, R>
+  export function go<L, R>(generator: () => Generator<FutureInstance<L, any>, R>): FutureInstance<L, R>
 
   /** Manage resources before and after the computation that needs them. See https://github.com/fluture-js/Fluture#hook */
   export function hook<L, H>(acquire: FutureInstance<L, H>): (dispose: (handle: H) => FutureInstance<any, any>) => <R>(consume: (handle: H) => FutureInstance<L, R>) => FutureInstance<L, R>
@@ -198,7 +183,7 @@ declare module 'fluture' {
       resolve: ResolveFunction<R>
     ) => Cancel): FutureInstance<L, R>
 
-    'fantasy-land/chainRec'<L, I, R>(iterator: (next: (value: I) => Next<I>, done: (value: R) => Done<R>, value: I) => FutureInstance<L, Next<I> | Done<R>>, initial: I): FutureInstance<L, R>
+    'fantasy-land/chainRec'<L, I, R>(iterator: (next: (value: I) => IteratorYieldResult<I>, done: (value: R) => IteratorReturnResult<R>, value: I) => FutureInstance<L, IteratorYieldResult<I> | IteratorReturnResult<R>>, initial: I): FutureInstance<L, R>
     'fantasy-land/of': typeof resolve
 
     '@@type': string

--- a/index.d.ts
+++ b/index.d.ts
@@ -183,7 +183,7 @@ declare module 'fluture' {
       resolve: ResolveFunction<R>
     ) => Cancel): FutureInstance<L, R>
 
-    'fantasy-land/chainRec'<L, I, R>(iterator: (next: (value: I) => IteratorYieldResult<I>, done: (value: R) => IteratorReturnResult<R>, value: I) => FutureInstance<L, IteratorYieldResult<I> | IteratorReturnResult<R>>, initial: I): FutureInstance<L, R>
+    'fantasy-land/chainRec'<L, I, R>(iterator: (next: (value: I) => IteratorYieldResult<I>, done: (value: R) => IteratorReturnResult<R>, value: I) => FutureInstance<L, IteratorResult<I, R>>, initial: I): FutureInstance<L, R>
     'fantasy-land/of': typeof resolve
 
     '@@type': string

--- a/package.json
+++ b/package.json
@@ -29,7 +29,7 @@
     "test:build": "npm run build && node --experimental-modules --no-warnings -- ./node_modules/.bin/oletus test/build/*.js",
     "coverage:upload": "c8 report --reporter=text-lcov > coverage.lcov && codecov",
     "coverage:report": "c8 report --reporter=html",
-    "test:types": "tsc --lib es6 index.d.ts"
+    "test:types": "tsc index.d.ts"
   },
   "author": "Aldwin Vlasblom <aldwin.vlasblom@gmail.com> (https://github.com/Avaq)",
   "homepage": "https://github.com/fluture-js/Fluture",
@@ -84,7 +84,7 @@
     "sanctuary-benchmark": "^1.0.0",
     "sanctuary-either": "^1.1.0",
     "sanctuary-type-classes": "^11.0.0",
-    "typescript": "^3.0.3",
+    "typescript": "^3.6.4",
     "xyz": "github:davidchambers/xyz#340d089"
   }
 }

